test/protocol.js

```diff
@@ -0,0 +1,176 @@
+        nonce
+        ;
+
+    async function getGas(func, name) {
+        let spent = await func.receipt.gasUsed
+        gas.push([name + ' gas: ', spent])
+    }
+
+    async function getBal(contract, address, name, units) {
+        let bal = (await web3.utils.fromWei((await contract.balanceOf(address)).toString()));
+        console.log(`${name} has in bank:`, await web3.utils.fromWei(await _prime.getBalance(Alice, contract.address)))
+        console.log(`${name} has a balance of ${bal} ${units}.`);
+    }
+
+    async function getEthBal(address, name, units) {
+        let bal = await web3.utils.fromWei((await web3.eth.getBalance(address)).toString());
+        console.log(`${name} has a balance of ${bal} ${units}.`);
+    }
+
+    async function getPoolBalances(poolInstance) {
+        let liability = await web3.utils.fromWei(await poolInstance._liability());
+        let lp1Funds = await web3.utils.fromWei(await poolInstance.balanceOf(Alice));
+        let lp1Ether = await web3.utils.fromWei((await web3.eth.getBalance(Alice)).toString());
+        let lp2Funds = await web3.utils.fromWei(await poolInstance.balanceOf(Bob));
+        let lp2Ether = await web3.utils.fromWei((await web3.eth.getBalance(Bob)).toString());
+        let totalDeposit = await web3.utils.fromWei(await poolInstance.totalSupply());
+        console.log({/* pool, */ liability, lp1Funds, lp1Ether, lp2Funds, lp2Ether, totalDeposit})
+        return ({/* pool, */ liability, lp1Funds, lp1Ether, lp2Funds, lp2Ether, totalDeposit});
+    }
+
+    beforeEach(async () => {
+
+        _prime = await Prime.deployed();
+        _tUSD = await tUSD.deployed();
+        _tETH = await tETH.deployed();
+        _exchange = await Exchange.deployed();
+        _pool = await Pool.deployed();
+        collateral = await web3.utils.toWei('1');
+        payment = await web3.utils.toWei('10');
+        collateralPoolAddress = _pool.address;
+        strikeAddress = _tUSD.address;
+        primeAddress = await _exchange.getPrimeAddress();
+        expiration = '1587607322';
+        premium = (13*10**16).toString();
+    });
+    
+
+    it('create prime with yak == erc20', async () => {
+        /* Base Parameters */
+        let minter = Bob;
+        let one = await web3.utils.toWei('1');
+        let ten = await web3.utils.toWei('10');
+        let collateral = _tETH.address();
+        let strike = _tUSD.address();
+        let expiry = '1587607322';
+        let receiver = minter;
+
+        /* Get revert with balance < xis */
+        await truffleAssert.reverts(
+            _prime.createPrime(
+                one,
+                collateral,
+                ten,
+                strike,
+
+            ),
+            "a < b"
+        );
+        /* Check for ERC20 interface parameter */
+
+        /* Assert liability of msg sender added the xis */
+
+        /* Assert returns the nonce of the token */
+    });
+
+    it('should be able to withdraw funds', async () => {
```

Comment:
Why do we have so many of the same test here? And why are they empty?

Reply:
was building this while we were talking about prs so I just pushed all my changes so I can start making prs